subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c

```diff
@@ -1336,6 +1335,31 @@ static void isr_done(void *param)
 	}
 }
 
+static uint8_t payload_index_get(const struct lll_sync_iso *lll)
+{
+	uint16_t payload_index;
+
+	if (lll->ptc_curr) {
+		uint8_t ptx_idx = lll->ptc_curr - 1U; /* max. nse 5 bits */
+		uint8_t ptx_payload_idx;
+		uint16_t ptx_group_mult;
+		uint8_t ptx_group_idx;
+
+		/* Calculate group index and multiplier for deriving
+		 * pre-transmission payload index.
+		 */
+		ptx_group_idx = ptx_idx / lll->bn; /* 5 bits */
+		ptx_payload_idx = ptx_idx - ptx_group_idx * lll->bn; /* 8 bits */
+		ptx_group_mult = (ptx_group_idx + 1U) * lll->pto; /* 9 bits */
+		payload_index = ptx_payload_idx + ptx_group_mult * lll->bn; /* 13 bits */
+		LL_ASSERT(payload_index <= UINT8_MAX);
+	} else {
+		payload_index  = lll->bn_curr - 1U; /* 3 bits */
+	}
+
+	return payload_index & 0xFF;
```

Comment:
If you want to ensure that you return the correct type, you should do this:

```suggestion
	return (uint8_t)payload_index;
```

`0xFF` is an `int` so you are effectively implicitly casting the type to an `int` here, and you already have an assert for the correct value range above

Reply:
Updated code. ISO Sync Receiver has no data type limitation that is applicable for ISO Broadcaster wrt buffer indices (runtime check for overflows are present).